src/agent/onefuzz-task/src/tasks/coverage/generic.rs

```diff
@@ -245,25 +235,30 @@ impl<'a> TaskContext<'a> {
 
     async fn try_record_input(&mut self, input: &Path) -> Result<()> {
         let coverage = self.record_impl(input).await?;
-        self.coverage.merge_max(&coverage);
+        self.coverage.merge(&coverage);
 
         Ok(())
     }
 
-    async fn record_impl(&mut self, input: &Path) -> Result<CommandBlockCov> {
-        let cache = Arc::clone(&self.cache);
-        let filter = self.filter.clone();
+    async fn record_impl(&mut self, input: &Path) -> Result<BinaryCoverage> {
+        let allowlist = self.allowlist.clone();
         let cmd = self.command_for_input(input).await?;
         let timeout = self.config.timeout();
-        let coverage = spawn_blocking(move || {
-            let mut cache = cache
-                .lock()
-                .map_err(|_| format_err!("module cache mutex lock was poisoned"))?;
-            record_os_impl(cmd, timeout, &mut cache, filter)
+        let recorded = spawn_blocking(move || {
+            CoverageRecorder::new(cmd)
+                .allowlist(allowlist)
+                .timeout(timeout)
+                .record()
         })
         .await??;
 
-        Ok(coverage)
+        if let Some(status) = recorded.output.status {
+            if !status.success() {
```

Comment:
Offline suggestion from @Porges: check for error code that matches an unhandled debug break on Windows, yield a more-specific error to aid debugging.

Reply:
This extra scrutiny is no longer necessary: such runtime faults are directly converted to to recording errors in #2772.